documentation/projects/proposals/project_improvement/20230914-implementation_plan_project_automations.md

```diff
@@ -0,0 +1,262 @@
+
+2. Column changes
+
+   - **Issues**
+     - Create new column "🗑️ Discarded" for issues that were closed without
+       resolution (like label "⛔️ status: discarded").
+   - **PRs**
+     - Split "Needs Review" into two columns based on their number of existing
+       reviews.
+       - 2️⃣ Needs 2 Reviews (no reviews yet)
+       - 1️⃣ Needs 1 Review (one approval, near finish line)
+     - Create new column "🗑️ Discarded" for PRs that were closed without merge.
+     - Rename "Merged" to "✅ Done" for consistency with issues board.
+
+### Improvements to existing automations
+
+1. [Issues/Issue is created](/meta/project_boards/issues.md#issue-is-created)
+   and
+   [Issues/Issue is added to the project](/meta/project_boards/issues.md#issue-is-added-to-the-project)
+
+   The issue should be added to the project board under the appropriate column,
+   "📋 Backlog" or "⛔ Blocked" depending on the issue labels.
+
+2. [Issues/Issue is closed](/meta/project_boards/issues.md#issue-is-closed)
+
+   The issue should be moved to the "✅ Done" column only if it was resolved. If
+   it was closed as discarded, it should be moved to a new "🗑️ Discarded"
+   column.
+
+   The built-in workflow will have to be replaced by a custom one that looks at
+   the `state_reason` field inside the issue payload of the `closed` activity of
+   the
+   [`issues` event](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#issues).
+
+3. [Issues/Issue is closed and inactive](/meta/project_boards/issues.md#issue-is-closed-and-inactive)
+
+   The current system means that issues are archived one by one as they become
+   inactive and with an 8 day cadence, this will slowly drift out of sync with
+   the weekly development chat.
+
+   Archiving should be performed by a custom scheduled workflow that archives
+   all items in the "✅ Done" and "🗑️ Discarded" columns weekly (preferably
+   immediately after the chat).
+
+4. [PRs/PR is created](/meta/project_boards/prs.md#pr-is-created) and
+   [PRs/PR is added to project](/meta/project_boards/prs.md#pr-is-added-to-the-project)
+
+   The PR should be added to the project board under the appropriate column, "In
+   Progress" or "Needs &lt;x&gt; Review(s)" depending on the PR's draft or
+   ready-for-review state and the number of existing reviews.
+
+5. [PRs/PR is closed or merged](/meta/project_boards/prs.md#pr-is-closed-or-merged)
+
+   The PR should be moved to the "✅ Done" column only if it was merged. If it
+   was closed without merge, it should be moved to a new "🗑️ Discarded" column.
+
+   This can be implemented by updating built-in workflows.
+
+### Restoration of previous automations
+
+1. Issues/Issue is connected to PR
+
+   When an issue is
+   [connected](https://docs.github.com/en/rest/overview/issue-event-types?apiVersion=2022-11-28#connected)
+   to a PR, the issue should move from "📋 Backlog" or "📅 To do" to the "🏗 In
```

Comment:
Just curious - would it make sense to say that it should move from _any_ "open" state to "In Progress"? For instance, should we stop a blocked issue from moving into "in progress" if a PR is opened for it?

Reply:
I agree with you that the card should move to "In progress" from any column if it indeed has progress made towards it. In that case it is a case of incorrect labelling and we can correct it.

Reply:
Fixed in 2f61873.